service/main.py

```diff
@@ -1,9 +1,50 @@
-from fastapi import FastAPI, File, UploadFile
+from fastapi import FastAPI, File, UploadFile, HTTPException
 from fastapi.responses import JSONResponse
+from logger import logger
+from dotenv import load_dotenv
+import tempfile
+import os
+# Import configurations and functions from modules
+from load_model import load_model
+from audio_service import convert_uploadfile_to_ndarray, translate_with_whisper,summarize_using_openai
 
 app = FastAPI()
 
+
+@app.get("/")
+def root_route():
+    return 'Hello, this is the root route for lingo ai server'
+
 @app.post("/upload-audio")
-def upload_audio(file: UploadFile = File(...)):
-    return {"filename": file.filename, "message": "File uploaded successfully!"}
+async def upload_audio(audioFile: UploadFile = File(...)):
+    try:
+
+        # Check file type
+        if not audioFile.filename.endswith(('m4a', 'mp4','mp3','webm','mpga','wav','mpeg')):
+            logger.error("invalid file type")
+            raise HTTPException(status_code=400, detail="Invalid file type")
+
+        # Create a temporary file to save the uploaded file
+        with tempfile.NamedTemporaryFile(delete=False, suffix=os.path.splitext(audioFile.filename)[1]) as temp_file:
+            temp_file_path = temp_file.name
+            # Save the file
+            with open(temp_file_path, "wb") as buffer:
+                buffer.write(await audioFile.read())
+
+        #translation = translate_with_ollama(transcription)
+        translation = translate_with_whisper(temp_file_path)
```

Comment:
can we pass the uploaded file path directly to translate_with_whisper() instead of saving temp?

Reply:
It saves the temp file and passes the temp file path to the translate_with_whisper as of now. I will try look for an alternative if any